src/base/tests/test_util.py

```diff
@@ -1057,6 +1057,43 @@ def test_create_m2m(self):
         self.assertEqual("res_groups_res_users_rel", auto_generated_m2m_table_name)
         self.assertTrue(util.table_exists(cr, auto_generated_m2m_table_name))
 
+    def test_rename_m2m(self):
+        cr = self.env.cr
+        partner_model_id = self.env["ir.model"].search([("model", "=", "res.partner")]).id
+
+        field_regular = self.env["ir.model.fields"].create(
+            {
+                "name": "x_m2m_field_regular",
+                "ttype": "many2many",
+                "model_id": partner_model_id,
+                "relation": "res.currency",
+            }
+        )
+        old_regular_table = field_regular.relation_table
+        self.assertEqual(old_regular_table, "x_res_currency_res_partner_rel")
+        field_custom = self.env["ir.model.fields"].create(
+            {
+                "name": "x_m2m_field_custom",
+                "ttype": "many2many",
+                "model_id": partner_model_id,
+                "relation": "res.currency",
+                "relation_table": "x_res_currency_res_partner_rel_2",
+            }
+        )
+        old_custom_table = field_custom.relation_table
+
+        util.rename_model(cr, "res.currency", "res.zurrency")
```

Comment:
I had to use `rename_model` instead of directly `update_m2m_tables` because the latter assumes a table is renamed and gets the m2m related to the new table name. I can alternatively call `rename_table` +  `update_m2m_tables`

Reply:
I would prefer. That would allow testing against injected models/fields instead of messing up with `res.currency`